lib/smart_city/events.ex

```diff
@@ -0,0 +1,19 @@
+defmodule SmartCity.Events do
+  @moduledoc """
+  Defines macros for encoding event types the Smart City platform
+  will respond to in any of the various micro service components
+  in a central location shared by all components.
+  """
+
+  @doc """
+  Defines an update event to a dataset within the system. The
+  system treats create events as a subset of updates.
+  """
+  defmacro update_dataset(), do: "dataset:update"
```

Comment:
Why do these need to be macros as opposed to just functions?

Reply:
So they can be used to pattern match in handle_event functions.

Reply:
Yeah, you can’t use functions as a pattern matcher in a function head so the macro just evaluates to that string at compile time.

Reply:
Ah! That makes sense.